clickhouse-admin/tests/integration_test.rs

```diff
@@ -2,130 +2,89 @@
-use omicron_test_utils::dev::test_setup_log;
-use oximeter_test_utils::wait_for_keepers;
-use slog::info;
+use slog::{info, o, Drain};
+use slog_term::{FullFormat, PlainDecorator, TestStdoutWriter};
 use std::collections::BTreeSet;
 use std::net::{Ipv6Addr, SocketAddrV6};
 use std::str::FromStr;
 
+fn log() -> slog::Logger {
+    let decorator = PlainDecorator::new(TestStdoutWriter);
+    let drain = FullFormat::new(decorator).build().fuse();
+    let drain = slog_async::Async::new(drain).build().fuse();
+    slog::Logger::root(drain, o!())
+}
+
+// In Clickward, keeper server ports are assigned by adding i to each
+// base port. Keeper IDs are also assigned with consecutive numbers
+// starting with 1.
+fn get_keeper_server_port(keeper_id: KeeperId) -> u16 {
+    let raw_id = keeper_id.0;
+    // We can safely unwrap raw_id as the Keeper IDs we use for testing are
+    // all in the single digits
+    DEFAULT_CLICKHOUSE_ADMIN_BASE_PORTS.keeper + u16::try_from(raw_id).unwrap()
+}
+
+fn get_keeper_raft_port(keeper_id: KeeperId) -> u16 {
+    let raw_id = keeper_id.0;
+    DEFAULT_CLICKHOUSE_ADMIN_BASE_PORTS.raft + u16::try_from(raw_id).unwrap()
+}
+
 #[tokio::test]
 async fn test_lgif_parsing() -> anyhow::Result<()> {
-    let logctx = test_setup_log("test_lgif_parsing");
-    let log = logctx.log.clone();
-
-    let (parent_dir, prefix) = log_prefix_for_test(logctx.test_name());
-    let path = parent_dir.join(format!("{prefix}-oximeter-clickward-test"));
-    std::fs::create_dir(&path)?;
-
-    // We spin up several replicated clusters and must use a
-    // separate set of ports in case the tests run concurrently.
-    let base_ports = BasePorts {
-        keeper: 29000,
-        raft: 29100,
-        clickhouse_tcp: 29200,
-        clickhouse_http: 29300,
-        clickhouse_interserver_http: 29400,
-    };
-
-    let config = DeploymentConfig {
-        path: path.clone(),
-        base_ports,
-        cluster_name: "oximeter_cluster".to_string(),
-    };
-
-    let mut deployment = Deployment::new(config);
-
-    // We only need a single keeper to test the lgif command
-    let num_keepers = 1;
-    let num_replicas = 1;
-    deployment
-        .generate_config(num_keepers, num_replicas)
-        .context("failed to generate config")?;
-    deployment.deploy().context("failed to deploy")?;
-
-    wait_for_keepers(&log, &deployment, vec![clickward::KeeperId(1)]).await?;
+    let log = log();
 
     let clickhouse_cli = ClickhouseCli::new(
-        Utf8PathBuf::from_str("clickhouse").unwrap(),
-        SocketAddrV6::new(Ipv6Addr::LOCALHOST, 29001, 0, 0),
+        Utf8PathBuf::from_str("clickhouse")?,
+        SocketAddrV6::new(
+            Ipv6Addr::LOCALHOST,
+            get_keeper_server_port(KeeperId(1)),
+            0,
+            0,
+        ),
     )
-    .with_log(log.clone());
+    .with_log(log);
 
-    let lgif = clickhouse_cli.lgif().await.unwrap();
+    let lgif = clickhouse_cli.lgif().await?;
```

Comment:
It's totally fine to use `?` here, but using unwrap will give you the proper line number for the failure, which may help debugging later.

Reply:
Huh, I wonder why I changed this. I didn't change the `.unwrap()` for a `?` anywhere else 😄